src/execution/dispatcher.js

```diff
@@ -35,116 +35,196 @@ export type ExecutionResult = {|
  *   - `hasNext` is true if a future payload is expected.
  */
 export type ExecutionPatchResult = {|
+  errors?: $ReadOnlyArray<GraphQLError>,
+  data?: ObjMap<mixed> | mixed | null,
+  path?: $ReadOnlyArray<string | number>,
+  label?: string,
+  extensions?: ObjMap<mixed>,
+  hasNext: boolean,
```

Comment:
I made a separate type because `ExecutionPatchResult` gets exported at the root of graphql-js and is used in return type of the main graphql function. In those cases, `hasNext` must always be defined so I don't think it should be optional. `DispatcherResult` is only ever used in private functions in the dispatcher.